File: rust/others/src/bin/abc198_c.rs
use proconio::input;

fn main() {
    input! {
        r:i128,
        x:i128,
        y:i128
    }
    let dd:i128 = x * x + y * y;
    let rr:i128 = r * r;
    if dd < rr {
        println!("2");
    } else {
        let mut ok: i128 = 200000;
        let mut ng: i128 = 0;
        while (ok - ng).abs() > 1 {
            let mid:i128 = (ok + ng) / 2;
            if mid * mid * rr >= dd {
                ok = mid;
            } else {
                ng = mid;
            }
        }
        println!("{}", ok);
    }
}

File: python/abc-c/abc198_c.py
# from math import *
# r,x,y=map(int,input().split())
# d=sqrt(x*x+y*y)
# if d<r:
#     print(2)
# else:
#     print(ceil(d/r))
r, x, y = map(int, input().split())
dd, rr = x * x + y * y, r * r
if dd < rr:
    print(2)
else:
    ok = 200000
    ng = 0
    while abs(ok - ng) > 1:
        mid = (ok + ng) // 2
        if mid ** 2 >= dd / rr:
            ok = mid
        else:
            ng = mid
    print(ok)
